### PR TITLE
migrate capi 1.5 jobs

### PR DESCRIPTION
**Pull request by kranurag7**
This PR migrates cluster-api 1.5 jobs to community cluster.

Part of https://github.com/kubernetes-sigs/cluster-api/issues/8689

### CODE REVIEW

**Comment by sbueringer** [comment]
Part of #8689

(please add ^^ to the PR description)

File: config/jobs/kubernetes-sigs/cluster-api/cluster-api-presubmits-release-1-5.yaml [hunk 1]
```diff
@@ -291,12 +313,11 @@ presubmits:
...
+    cluster: eks-prow-build-cluster
     labels:
       preset-dind-enabled: "true"
       preset-kind-volume-mounts: "true"
     decorate: true
-    decoration_config:
-      gcs_credentials_secret: "" # Use workload identity for uploading artifacts
     always_run: false
     branches:
     # The script this job runs is not in all branches.
```
Comment (sbueringer) [inline config/jobs/kubernetes-sigs/cluster-api/cluster-api-presubmits-release-1-5.yaml:327]: I think we're missing something in l.309++

Comment (sbueringer) [reply config/jobs/kubernetes-sigs/cluster-api/cluster-api-presubmits-release-1-5.yaml:327]: Same in L.353++ and l.389++

File: config/jobs/kubernetes-sigs/cluster-api/cluster-api-periodics-release-1-5.yaml [hunk 2]
```diff
@@ -104,16 +106,19 @@ periodics:
         resources:
           requests:
             cpu: 7300m
+            memory: 9Gi
+          limits:
+            cpu: 7300m
+            memory: 9Gi
```
Comment (sbueringer) [inline config/jobs/kubernetes-sigs/cluster-api/cluster-api-periodics-release-1-5.yaml:112]:
```suggestion
            memory: 32Gi
          limits:
            cpu: 7300m
            memory: 32Gi
```
Let's please use the same values as on release-1.4

File: config/jobs/kubernetes-sigs/cluster-api/cluster-api-periodics-release-1-5.yaml [hunk 3]
```diff
@@ -149,16 +153,19 @@ periodics:
...
```
Comment (sbueringer) [inline config/jobs/kubernetes-sigs/cluster-api/cluster-api-periodics-release-1-5.yaml:159]:
```suggestion
            memory: 32Gi
          limits:
            cpu: 7300m
            memory: 32Gi
```

File: config/jobs/kubernetes-sigs/cluster-api/cluster-api-periodics-release-1-5.yaml [hunk 4]
```diff
@@ -195,6 +201,10 @@ periodics:
       resources:
         requests:
           cpu: 7300m
+          memory: 9Gi
+        limits:
+          cpu: 7300m
+          memory: 9Gi
```
Comment (sbueringer) [inline config/jobs/kubernetes-sigs/cluster-api/cluster-api-periodics-release-1-5.yaml:207]:
```suggestion
          memory: 32Gi
        limits:
          cpu: 7300m
          memory: 32Gi
```

File: config/jobs/kubernetes-sigs/cluster-api/cluster-api-periodics-release-1-5-upgrades.yaml [hunk 5]
```diff
@@ -47,17 +45,20 @@ periodics:
       resources:
         requests:
           cpu: 7300m
+          memory: 9Gi
```
Comment (sbueringer) [inline config/jobs/kubernetes-sigs/cluster-api/cluster-api-periodics-release-1-5-upgrades.yaml:48]: Please compare the entire file with release-1.4 and use the same resources

File: config/jobs/kubernetes-sigs/cluster-api/cluster-api-presubmits-release-1-5.yaml [hunk 6]
```diff
@@ -153,23 +167,25 @@ presubmits:
         resources:
           requests:
             cpu: 7300m
+            memory: 9Gi
+          limits:
+            cpu: 7300m
+            memory: 9Gi
```
Comment (sbueringer) [inline config/jobs/kubernetes-sigs/cluster-api/cluster-api-presubmits-release-1-5.yaml:173]:
```suggestion
            memory: 32Gi
          limits:
            cpu: 7300m
            memory: 32Gi
```
(e2e test job should use the same resources as the e2e periodics)

File: config/jobs/kubernetes-sigs/cluster-api/cluster-api-presubmits-release-1-5.yaml [hunk 7]
```diff
@@ -184,24 +200,26 @@ presubmits:
...
```
Comment (sbueringer) [inline config/jobs/kubernetes-sigs/cluster-api/cluster-api-presubmits-release-1-5.yaml:206]:
```suggestion
            memory: 32Gi
          limits:
            cpu: 7300m
            memory: 32Gi
```

File: config/jobs/kubernetes-sigs/cluster-api/cluster-api-presubmits-release-1-5.yaml [hunk 8]
```diff
@@ -216,23 +234,25 @@ presubmits:
...
```
Comment (sbueringer) [inline config/jobs/kubernetes-sigs/cluster-api/cluster-api-presubmits-release-1-5.yaml:240]:
```suggestion
            memory: 32Gi
          limits:
            cpu: 7300m
            memory: 32Gi
```

File: config/jobs/kubernetes-sigs/cluster-api/cluster-api-presubmits-release-1-5.yaml (hunk 1 above)
Comment (kranurag7) [reply config/jobs/kubernetes-sigs/cluster-api/cluster-api-presubmits-release-1-5.yaml:327]: @sbueringer I force pushed little early. I'll add additional commits and will squash later. 

I have used 32Gi for all e2e jobs in the latest commit. 
I've made sure that there's no usage of `decorate_config` or `gcs_credentials_secret` in the jobs.

**Comment by sbueringer** [comment]
Thx!

/lgtm
/approve

/hold

**Comment by nawazkh** [comment]
Kudos on leading this effort @kranurag7 
/lgtm

**Pull request merged**